src/state/controls/Button.ts

```diff
@@ -71,4 +71,11 @@ export class Button extends Control<IButtonData> implements IButton {
     public giveInput(input: IButtonInput): Promise<void> {
         return this.sendInput(input);
     }
+
+    public update(changedData: Partial<IButtonData>): Promise<void> {
+        if (changedData.cooldown) {
```

Comment:
It would be preferable to use an entirely separate method, then, which takes a data type that is specifically documented to be different! I would also make it have a `cooldownDuration` instead of `cooldown` property, to ensure that (at least TS) consumers can't incorrectly swap the methods, and to prevent general confusion between the data types.